specs/sharding/beacon-chain.md

```diff
@@ -486,18 +512,8 @@ def get_shard_proposer_index(beacon_state: BeaconState, slot: Slot, shard: Shard
     """
     epoch = compute_epoch_at_slot(slot)
     seed = hash(get_seed(beacon_state, epoch, DOMAIN_SHARD_BLOB) + uint_to_bytes(slot) + uint_to_bytes(shard))
-
-    # Proposer must have sufficient balance to pay for worst case fee burn
-    EFFECTIVE_BALANCE_MAX_DOWNWARD_DEVIATION = (
-        EFFECTIVE_BALANCE_INCREMENT - EFFECTIVE_BALANCE_INCREMENT
-        * HYSTERESIS_DOWNWARD_MULTIPLIER // HYSTERESIS_QUOTIENT
-    )
-    min_effective_balance = (
-        beacon_state.shard_gasprice * MAX_SAMPLES_PER_BLOB // TARGET_SAMPLES_PER_BLOB
-        + EFFECTIVE_BALANCE_MAX_DOWNWARD_DEVIATION
-    )
```

Comment:
Why don't we check the balance any more?
Intuitively an active validator should have enough balance to pay for blob with any 'reasonable' price, but anyways that check was originally here for some reason

Reply:
We should be checking the builder balance (TODO), not the shard proposer balance (they don't burn with their own balance, the builder does). Nice side-effect: any active validator can be a shard-proposer, no exceptions.